OJS.Workers.Checkers/CSharpCodeChecker.cs

```diff
@@ -1,47 +1,61 @@
-﻿namespace OJS.Workers.Checkers
+﻿using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.Emit;
+using System.Reflection;
+using System.Text.RegularExpressions;
+
+namespace OJS.Workers.Checkers
 {
     using System;
-    using System.CodeDom.Compiler;
     using System.Text;
-
-    using Microsoft.CSharp;
     using OJS.Workers.Checkers.CSharpCodeCheckers;
 
     public class CSharpCodeChecker
         : CSharpCodeCheckerBase
     {
         protected override Type CompileCheckerAssembly(string sourceCode)
         {
-            var codeProvider = new CSharpCodeProvider();
-            var compilerParameters = new CompilerParameters { GenerateInMemory = true, };
-            compilerParameters.ReferencedAssemblies.Add("System.dll");
-            compilerParameters.ReferencedAssemblies.Add("System.Core.dll");
-            compilerParameters.ReferencedAssemblies.Add("OJS.Workers.Common.dll");
-            var compilerResults = codeProvider.CompileAssemblyFromSource(compilerParameters, sourceCode);
-
-            this.CheckForErrors(compilerResults);
-            return this.GetCustomCheckerType(compilerResults.CompiledAssembly);
-        }
+           var syntaxTree = CSharpSyntaxTree.ParseText(sourceCode);
+            var assemblyPath = Path.GetDirectoryName(typeof(object).Assembly.Location);
 
-        private void CheckForErrors(CompilerResults compilerResults)
-        {
-            if (!compilerResults.Errors.HasErrors)
+            var references = new List<MetadataReference>
             {
-                return;
-            }
+                MetadataReference.CreateFromFile(typeof(object).Assembly.Location),
+                MetadataReference.CreateFromFile(typeof(Enumerable).Assembly.Location),
+                MetadataReference.CreateFromFile(typeof(Regex).Assembly.Location),
+                MetadataReference.CreateFromFile(Path.Combine(assemblyPath, "System.Runtime.dll")),
+                MetadataReference.CreateFromFile("OJS.Workers.Common.dll"),
```

Comment:
Can't you use Location for those as well?
For OJS should be easy - `typeOf(OJS.Workers.Common.SomeType).Assembly.Location`